### DISCUSSION TITLE
Chest Protection with "RegionProtectChests"

### DISCUSSION THREAD

**Discussion by iwickdev** [Pryaxis/TShock#1972] [Q&A] [answered]
In the TShock config there is a setting called RegionProtectChests that says it is supposed to Protect chests with region and build permissions. I changed this to True and restarted the server but it does not seem to be doing anything, when I place down a chest I don't see any region or protection being setup that stops users from opening the chest. I also tested this with different users and accounts but nothing seemed to change.
Is there a type of permission I need before players can use it?

Thanks

**Comment by hakusaro** [comment]
This could be a problem but someone needs to test to reproduce and verify it. It's probably a defect.

**Reply by iwickdev** [reply]
That is what I was thinking as well. No error message or anything in console, just nothing happens.

**Reply by moisterrific** [reply]
Just tested it, `RegionProtectChests = true` is working perfectly fine on my end.

Not sure why it doesn't work for you, maybe some plugin causing it?

Test methods:
1. delete \tshock\ folder to allow it to generate a new one
2. delete every plugin in ServerPlugins except TShockAPI
3. boot up server, verify no other plugins are running
4. change `RegionProtectChests` in config to `true`
5. `/reload` through console to set the change
6. join as owner account and place a chest
7. define a protected region around the chest 
8. join as second test account and try to open chest
9. it said in red warning text "You lack permission to build in this region." when my test account tried to open the chest

Tested on 4.4.0 pre-10

**Comment by moisterrific** [comment, marked as answer]
> when I place down a chest I don't see any region or protection being setup that stops users from opening the chest. I also tested this with different users and accounts but nothing seemed to change.

Just reading this message over again I think you might be misunderstanding what that config option is for. `RegionProtectChests` is meant to protect chests that were placed in a protected region. It doesn't automatically generate a protected region when you place the chest, you have to define a protected region via `/region`. Only then if you set `RegionProtectChest = true`, unauthorized users/groups will not be able to open or remove that chest within the protected region.

**Reply by iwickdev** [reply]
Is there a limit on region Sizes we can change?
